src/pdn/src/PdnGen.tcl

```diff
@@ -3614,13 +3676,69 @@ proc generate_stripes {tag net_name} {
           set template_name [dict get $plan_template $x $y]
           set layer_info [dict get $grid_data straps $lay $template_name]
           set area [list $x $y [expr $x + [dict get $template width]] [expr $y + [dict get $template height]]]
-          generate_upper_metal_mesh_stripes $tag $lay $layer_info $area
+          generate_upper_metal_mesh_stripes $tag $lay $layer_info $area $net_name
         }
       }
     }
   }
 }
 
+proc insert_power_switches {} {
+  variable grid_data
+
+  set db [ord::get_db]
+  set block [ord::get_db_block]
+
+  set psw [$db findMaster [dict get $grid_data switch_cell]]
+  # The selected power switch is double height, and has a central VSS pin, so align with VSS rails
+  set vgnd [[$psw findMTerm VGND] getMPins]
+  set vddg [[$psw findMTerm VDDG] getMPins]
+
+  set rail_lay [lindex [get_rails_layers] 0]
+  set rail_width [dict get $grid_data rails $rail_lay width]
+
+  set strap_lay [lindex [dict get $grid_data straps] 0]
+  set strap_lay_info [dict get $grid_data straps $strap_lay]
+  set strap_width [dict get $strap_lay_info width]
+
+  set area [dict get $grid_data area]
+  set offset [expr [lindex $area 0] + [dict get $strap_lay_info offset]]
+
+  set row_idx 0
+
+  foreach row [$block getRows] {
+    set orient [$row getOrient]
+    set box [$row getBBox]
+    if {$orient == "R0"} {
+      set vss_y [$box yMin]
+      set overlap_y [expr $vss_y - $rail_width / 2]
+      
+      set col_idx 0
+
+      for {set x $offset} {$x < [expr {[lindex $area 2] - [dict get $strap_lay_info width]}]} {set x [expr {[dict get $strap_lay_info pitch] + $x}]} {
+        set overlap_x [expr $x - $strap_width / 2]
+
+        set location [list \
+          [expr $overlap_x - [[$vddg getBBox] xMin]] \
+          [expr $overlap_y - [[$vgnd getBBox] yMin]] \
+        ]
+
+        set inst_name "PSW_${row_idx}_${col_idx}"
+        if {[set inst [$block findInst $inst_name]] == "NULL"} {
+          set inst [odb::dbInst_create $block $psw $inst_name]
+        }
```

Comment:
Does this assume the user would have put the switches in themselves? If so, you should check the type to make sure the masters match.

Reply:
The intent would be to ue instances if they are already present in the design. I dont have any particular reason to suspect this would be the case, but I agree that we should check the cell master if we want to use an instance returned by findInst.

Reply:
Since the intent is to use instance already in the design, then I think it is worth spending time to figure out the right way to insert these. At the moment, the regions is not honored, insertion doesn't handle existing placement,, additional connectivity needs to be addressed (enable conditions, etc), etc.